cmd/serverless-init/cloudservice/cloudrun.go

```diff
@@ -14,60 +14,68 @@ import (
 )
 
 const (
-	//nolint:revive // TODO(SERV) Fix revive linter
 	revisionNameEnvVar      = "K_REVISION"
 	ServiceNameEnvVar       = "K_SERVICE" // ServiceNameEnvVar is also used in the trace package
 	configurationNameEnvVar = "K_CONFIGURATION"
 	functionTypeEnvVar      = "FUNCTION_SIGNATURE_TYPE"
 	functionTargetEnvVar    = "FUNCTION_TARGET" // exists as a cloudrunfunction env var for all runtimes except Go
 )
 
+const (
+	// SpanNamespace is the namespace for the span
+	cloudRunService  = "gcr."
+	cloudRunFunction = "gcrfx."
+)
+
 var metadataHelperFunc = helper.GetMetaData
 
 // CloudRun has helper functions for getting Google Cloud Run data
 type CloudRun struct {
+	spanNamespace        string
```

Comment:
This works well also, but I’m curious if by making everything constants as I stated earlier, you could have stuck with the original `cloudRunFunctionMode` bool. This way you could use the `gcr` constants as default and switch to the `gcrfx` ones if `cloudRunFunctionMode=true`. I think this would be easier to read as well. Can you help explain why my suggestions wouldn’t work and why you chose this approach instead?